fix(STONEINTG-1087): retry longer to fetch App from ITS, and stop processing if not found

dheerajodha opened the pull request:
* There are some cases where the App and ITS are created almost at the same time.
* This causes a problem when the App is not found while the controllers reconcile ITS.
* To fix this, we increased the retry time from 10 ms to 10 seconds. This should give enough time to fetch the App.
* And if we're still unable to find App, we stop reconciling, since there's no point in reconciling an ITS without any parent.

## Maintainers will complete the following section

- [ ] Commit messages are descriptive enough ([hints](https://www.freecodecamp.org/news/how-to-write-better-git-commit-messages/))
- [ ] Code coverage from testing does not decrease and new code is covered ([check the PR coverage on codecov](https://app.codecov.io/gh/konflux-ci/integration-service/pulls))
- [ ] [Controllers diagrams](https://github.com/konflux-ci/integration-service/tree/main/docs) are updated when PR changes controllers code  (if applicable)

dheerajodha committed:
fix(STONEINTG-1087): longer wait for App

* There are some case where the App and ITS are created
  almost at the same time. This causes problem when the
  App is not found while the controllers reconcile ITS.
* To fix this, we increased the retry time from 10 ms
  to 10 seconds.
* This should give enough time to fetch the App.
* And if we're still unable to find App, we stop
  reconciling, since there's no point in reconciling
  an ITS without any parent.

Signed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha commented on internal/controller/scenario/scenario_controller.go:
```diff
@@ -84,6 +92,7 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 	})
 	if err != nil {
 		logger.Error(err, "Failed to get Application from the IntegrationTestScenario after retry", "application", scenario.Spec.Application)
+		return ctrl.Result{}, nil
```
Once we have retried long enough for the Application to exist in the cluster, and still couldn't find it, the operator will stop processing this ITS. But in that case, this ITS would be an orphan and dangle in the cluster. Do you think we should delete this ITS here?

dirgim replied:
Hm, not sure, maybe we can at least mark it as invalid?

dheerajodha commented on internal/controller/scenario/scenario_controller.go:
```diff
@@ -75,15 +77,30 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
+			logger.Error(err, "Failed to update Scenario as Invalid")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{}, nil
```
Because we continue processing in case the Application is not found, I think we can remove [this block of code](https://github.com/konflux-ci/integration-service/blob/main/internal/controller/scenario/scenario_adapter.go#L72-L84) from the scenario_adapter.go file, wdyt?

dirgim replied:
Yup, that makes sense to me.

dheerajodha replied:
Thanks! Removed

dheerajodha committed:
fix(STONEIN-1087): remove Adapter code for nil app

* Now that we're checking for nil App within the
  scenario_controller.go file,
* AND calling ContinueProcessing() in case of nil App,
* AND marking the Scenario as Invalid in that case,
* We don't need to check and do the same thing on
  scenario_adapter.go file as well, hence removing
  this code.

Signed-off-by: Dheeraj<djodha@redhat.com>

dheerajodha committed:
fix(STONEINTG-1087): add App to logger sooner

* We should add Application to the logger,
  within the controller.go file rather than
  adapter.go file

Signed-off-by: Dheeraj<djodha@redhat.com>

dirgim reviewed (approved):
LGTM

jsztuka reviewed (approved):
lgtm
